src/rt/sections_elf_shared.d

```diff
@@ -892,10 +900,18 @@ else static if (SharedDarwin) void scanSegments(mach_header* info, DSO* pdso)
 
 bool findImageHeaderForAddr(in void* addr, out ImageHeader result) nothrow @nogc
 {
-    version (OSX)
+    version (Darwin)
     {
-        result = _dyld_get_image_header_containing_address(addr);
-        return !!result;
+        Dl_info info;
+        const res = dladdr(addr, &info);
+
+        if (res == 0)
+            return false;
+
+        if (result)
+            *result = cast(ImageHeader) info.dli_fbase;
```

Comment:
This looks as if it shouldn't compile (`result` is an `ImageHeader` ref).

Reply:
Oh, the surrounding code has changed since I implemented this. I'll update.

Reply:
Done.